server/src/service-layer/controllers/StripeWebhook.ts

```diff
@@ -26,40 +27,45 @@ export class StripeWebhook extends Controller {
       console.error(err)
       return this.setStatus(401) // unauthorized request
     }
-    // console.log(event.type)
-    if (event.type === "payment_intent.succeeded") {
-      // console.log("[WEBHOOK] received: payment_intent.succeeded")
-      console.log(request.body)
-      const userService = new UserDataService()
-      const authService = new AuthService()
-      // Stripe PaymentIntent
-      const { id } = event.data.object
-      // Fetch the checkout session from the PaymentIntent ID
-      const stripeSession: Stripe.Checkout.Session =
-        await stripe.checkout.sessions.retrieve(id)
-      const uid = stripeSession.client_reference_id
+    // Create services
+    const userService = new UserDataService()
+    const authService = new AuthService()
+    const stripeService = new StripeService()
 
-      if (!uid || !(await userService.getUserData(uid)))
-        return this.setStatus(400) // bad request, non existent user
-      if (
-        event.data.object.metadata[CHECKOUT_TYPE_KEY] ===
-        CheckoutTypeValues.MEMBERSHIP
-      ) {
-        try {
-          // need to update firestore
-          await userService.editUserData(uid, {
-            membership: "member"
-          })
-          // need to add member claim to user
-          await authService.setCustomUserClaim(uid, "member")
-          console.log(
-            "successfully received checkout.session.completed and added membership"
+    switch (event.type) {
+      case "payment_intent.succeeded":
+        if (event.type === "payment_intent.succeeded") {
+          console.debug("[WEBHOOK] received payment_intent.succeeded")
+          // Stripe PaymentIntent
+          const { customer } = event.data.object
+          // Fetch the checkout session from the PaymentIntent ID
+          const stripeSession =
```

Comment:
```suggestion
          const [stripeSession] =
```

this will automatically give you the first element of the array (the array size is guaranteed to be at max 1